addons/binding/org.openhab.binding.tado/src/main/java/org/openhab/binding/tado/handler/TadoZoneHandler.java

```diff
@@ -0,0 +1,259 @@
+                return;
+            }
+
+            logger.debug("Updating state of home " + getHomeId() + " and zone " + getZoneId());
+
+            TadoZoneStateAdapter state = new TadoZoneStateAdapter(zoneState, getTemperatureUnit());
+            updateStateIfNotNull(TadoBindingConstants.CHANNEL_ZONE_CURRENT_TEMPERATURE, state.getInsideTemperature());
+            updateStateIfNotNull(TadoBindingConstants.CHANNEL_ZONE_HUMIDITY, state.getHumidity());
+            updateStateIfNotNull(TadoBindingConstants.CHANNEL_ZONE_HEATING_POWER, state.getHeatingPower());
+
+            updateState(TadoBindingConstants.CHANNEL_ZONE_OPERATION_MODE, state.getOperationMode());
+
+            updateState(TadoBindingConstants.CHANNEL_ZONE_HVAC_MODE, state.getMode());
+            updateState(TadoBindingConstants.CHANNEL_ZONE_TARGET_TEMPERATURE, state.getTargetTemperature());
+            updateState(TadoBindingConstants.CHANNEL_ZONE_FAN_SPEED, state.getFanSpeed());
+            updateState(TadoBindingConstants.CHANNEL_ZONE_SWING, state.getSwing());
+
+            updateState(TadoBindingConstants.CHANNEL_ZONE_TIMER_DURATION, state.getRemainingTimerDuration());
+
+            updateState(TadoBindingConstants.CHANNEL_ZONE_OVERLAY_EXPIRY, state.getOverlayExpiration());
+        } catch (IOException | TadoClientException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Could not connect to server due to " + e.getMessage());
+        }
+    }
+
+    private void scheduleZoneStateUpdate() {
+        if (refreshTimer == null || refreshTimer.isCancelled()) {
+            refreshTimer = scheduler.scheduleAtFixedRate(new Runnable() {
+                @Override
+                public void run() {
+                    updateZoneState(false);
+                }
+            }, 5, configuration.refreshInterval, TimeUnit.SECONDS);
+        }
+    }
+
+    private void cancelScheduledZoneStateUpdate() {
+        if (refreshTimer != null) {
+            refreshTimer.cancel(false);
+        }
+    }
+
+    private void scheduleHvacChange() {
+        if (scheduledHvacChange != null) {
+            scheduledHvacChange.cancel(false);
+        }
+
+        scheduledHvacChange = scheduler.schedule(() -> {
+            try {
+                TadoHvacChange change = this.pendingHvacChange;
+                this.pendingHvacChange = new TadoHvacChange(getThing());
+                change.apply();
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            } catch (TadoClientException e) {
+                logger.error("Could not apply HVAC change on home + " + getHomeId() + " and zone " + getZoneId() + ": "
+                        + e.getMessage(), e);
+            } finally {
+                updateZoneState(true);
+            }
+        }, configuration.hvacChangeDebounce, TimeUnit.SECONDS);
```

Comment:
Delaying the request is actually quite bad. A user expects that a value he selects will be set immediately and if not he might click on the respective action over and over again.

Also for your change to be send, can't you query those parameters which are currently set on the cloud side and put them into your change together with the changed value? This way you do not have to guess values that you do not know just to satisfy the cloud API.

And since you are working for Tado, did you think about changing the API on the cloud side to also accept requests which change only one value at a time?